src/Bugsnag/Notifier.cs

```diff
@@ -39,39 +42,71 @@ public void Send(Event errorEvent)
                 Send(notification);
         }
 
+        public void SendStoredReports()
+        {
+            try
+            {
+                foreach (var storedReport in Store.ReadStoredJson())
+                {
+                    SendJson(storedReport);
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.Warning(string.Format("Bugsnag failed to send persisted error reports with exception: {0}", e.ToString()));
+            }
+        }
+
         private void Send(Notification notification)
         {
             try
             {
-                //  Post JSON to server:
-                var request = WebRequest.Create(Config.EndpointUrl);
-
-                request.Method = WebRequestMethods.Http.Post;
-                request.ContentType = "application/json";
-                request.Proxy = DetectedProxy;
-
-                using (var streamWriter = new StreamWriter(request.GetRequestStream()))
-                {
-                    string json = JsonConvert.SerializeObject(notification, JsonSettings);
-                    streamWriter.Write(json);
-                    streamWriter.Flush();
-                }
-                request.GetResponse().Close();
+                SendJson(JsonConvert.SerializeObject(notification, JsonSettings));
             }
             catch (Exception e)
             {
-                Logger.Warning("Bugsnag failed to send error report with exception: " + e.ToString());
-
-                // Deliberate empty catch for now
-
-                // Must never double fault - i.e. can't leak an exception from an exception handler
-                // Also the caller might have just wanted to report an "information" bugsnag and we
-                // certainly don't want to fault then
-
-                // However, "offline" handling should be addressed. I.e. bugsnags should be queued
-                // until network is available and sent then, see
-                // https://github.com/bugsnag/bugsnag-dotnet/issues/31
+                Logger.Warning(string.Format("Bugsnag failed to serialise error report with exception: {0}", e.ToString()));
             }
         }
+
+        private void SendJson(string json)
+        {
+            var reportSent = false;
+            try
+            {
+                //  Post JSON to server:
+                var request = WebRequest.Create(Config.EndpointUrl);
+
+                request.Method = WebRequestMethods.Http.Post;
+                request.ContentType = "application/json";
+                request.Proxy = DetectedProxy;
+
+                using (var streamWriter = new StreamWriter(request.GetRequestStream()))
+                {
+                    streamWriter.Write(json);
+                    streamWriter.Flush();
+                }
+
+                using (request.GetResponse())
+                {

```

Comment:
No, this is basically the same as this [line](https://github.com/bugsnag/bugsnag-dotnet/blob/master/src/Bugsnag/Notifier.cs#L59). But in a using statement so that things are properly disposed